admin/utils.php

```diff
@@ -35,6 +35,110 @@ public static function on_edit_taxonomy( $term_id, $term ) {
 			}
 		}
 
+		/**
+		 * Loops through the critical CSS rules and makes requests
+		 * for any shared critical CSS values that need to be refreshed.
+		 * @author Jim Barnes
+		 * @since 0.1.0
+		 * @return void
+		 */
+		public static function update_shared_critical_css() {
+			$now = time();
+			$rules = get_field( 'ucfccss_deferred_rules', 'option' );
+
+			$value_key_lookup = array(
+				'post_type' => 'post_types',
+				'taxonomy'  => 'taxonomies',
+				'template'  => 'templates'
+			);
+
+			foreach( $rules as $rule ) {
+				if ( $rule['rule_type'] === 'shared' ) {
+
+					$value_key = isset( $value_key_lookup[$rule['object_type']] ) ?
+						$value_key_lookup[$rule['object_type']] :
+						null;
+
+					if ( ! $value_key ) continue;
+
+					foreach( $rule[$value_key] as $value ) {
+						// Returns the name of the rule we need to check.
+						$css_object_key = self::get_shared_rule_name( $rule['object_type'], $value );
+						$css_object_exp = "{$css_object_key}_expiration";
+
+						$css = get_option( $css_object_key, null );
```

Comment:
Because `get_field` triggers the `acf/init` action, and that can have some unintended consequences if it is called too early, I'd like to avoid using it for retrieving options for this project. In this case, it would probably not cause any issues, but I'd rather just be consistent throughout the project.